### PR TITLE
Add simplifications for Floating Point nodes

### PR DESCRIPTION
**Pull request by shivammittal99**
This PR is a part of my [GSOC'18 Project](https://summerofcode.withgoogle.com/projects/5175053981843456).

1. Added some simplifications for FP nodes : fadd, fsub, fmul, fneg, dadd, dsub, dmul, dneg.
2. Added Value Propagation constraints : VPFloatConstraint, VPFloatConst, VPFloatRange, VPDoubleConstriant, VPDoubleConst, VPDoubleRange ( #2667 )

### CODE REVIEW

**Comment by charliegracie** [comment]
@genie-omr build all

**Comment by 0xdaryl** [comment]
There appears to be a lot of symmetry between the float and double handlers (other than the occasional operation that does care about the type).  Can you restructure to share much of the logic and specialize on type only when necessary?  I'm concerned about code growth (footprint) and maintenance.

File: compiler/optimizer/OMRSimplifierHandlers.cpp [hunk 1]
```diff
@@ -6079,7 +6079,7 @@ TR::Node *anchorSimplifier(TR::Node * node, TR::Block * block, TR::Simplifier *
 //
 
 TR::Node *iaddSimplifier(TR::Node * node, TR::Block * block, TR::Simplifier * s)
-   {
+   {         
    simplifyChildren(node, block, s);
```
Comment (0dvictor) [inline compiler/optimizer/OMRSimplifierHandlers.cpp]: It might be better to trim trailing blanks.

**Comment by vijaysun-omr** [comment]
Can the value propagation changes be separated from the simplifier changes ? i.e. separate PR(s)

**Comment by vijaysun-omr** [comment]
There is a set of improvements to integer/long simplifiers in one of the commits too; I believe this is to add logic that was missing for them but was there (in part) for FP types ?

**Comment by 0xdaryl** [comment]
Is there any update to commoning up the very similar logic between the float and double handlers so that we can move this PR forward?

**Comment by vijaysun-omr** [comment]
I would appreciate a review from @andrewcraik on this too.

**Comment by vijaysun-omr** [comment]
Could you please comment on functional testing of this change ? 
Self contained changes such as tree simplifying in more cases ought to be possible to unit test reasonably.

**Comment by shivammittal99** [comment]
@vijaysun-omr I tested the changes by injecting TRIL code for each simplification and verifying the change in the logs generated. Although I couldn't find the appropriate location where such tests should go. Please guide.

**Comment by charliegracie** [comment]
ping @vijaysun-omr

**Comment by charliegracie** [comment]
@genie-omr build all

**Comment by DanHeidinga** [comment]
@vijaysun-omr @charliegracie @shivammittal99 When do you expect this will be ready to merge?  The builds from 11 days ago appear to have passed.

**Comment by vijaysun-omr** [comment]
I pinged @0xdaryl about this PR in case there were any discussions related to this PR with Shivam (Daryl was guiding this summer project on a week by week basis). If he knows of plans to deliver the tests that were alluded to earlier, then I would prefer making the decision to merge or not with that knowledge.

**Comment by 0xdaryl** [comment]
@shivammittal99 , if you have Tril tests the logical place for them to go is `fvtest/compilertriltest` into some kind of file like `SimplifierFoldFloatingPoint.cpp`.  I'm not sure I like the current precedent of creating multiple files for each kind of simplifier operation, but I don't think that needs to be decided in this PR.  Please contribute what you have.

**Comment by vijaysun-omr** [comment]
@shivammittal99 any update ?

**Comment by vijaysun-omr** [comment]
@0xdaryl any suggestions on how to proceed with this PR ?

**Comment by 0xdaryl** [comment]
I am closing this PR now given its age and the lack of responses for updates from committers.  Thanks for the contribution.

**Pull request closed without merging**